src/main/scala/mesosphere/marathon/core/instance/InstanceStatus.scala

```diff
@@ -86,4 +91,7 @@ object InstanceStatus {
   // Unknown: the master has no knowledge of the task
   case object Unknown extends InstanceStatus with Terminal
 
+  object Terminal {
+    def unapply(status: InstanceStatus): Option[InstanceStatus] = if (status.isTerminal) Some(status) else None
```

Comment:
Not sure why this is scary: this will match all terminal states. So you can say:
```scala
state match {
   case Terminal(term) => //state is terminal
   case _ => //non terminal
}
```
This is especially useful to combine this matcher.